termux-shared/src/main/java/com/termux/shared/file/FileUtils.java

```diff
@@ -0,0 +1,1627 @@
+     * @return Returns {@code true} if file exists, otherwise {@code false}.
+     */
+    public static boolean fileExists(final String filePath, final boolean followLinks) {
+        return getFileType(filePath, followLinks) != FileType.NO_EXIST;
+    }
+
+    /**
+     * Checks the type of file that exists at {@code filePath}.
+     *
+     * This function is a wrapper for
+     * {@link FileTypes#getFileType(String, boolean)}
+     * 
+     * @param filePath The {@code path} for file to check.
+     * @param followLinks The {@code boolean} that decides if symlinks will be followed while
+     *                       finding type. If set to {@code true}, then type of symlink target will
+     *                       be returned if file at {@code filePath} is a symlink. If set to
+     *                       {@code false}, then type of file at {@code filePath} itself will be
+     *                       returned.
+     * @return Returns the {@link FileType} of file.
+     */
+    public static FileType getFileType(final String filePath, final boolean followLinks) {
+        return FileTypes.getFileType(filePath, followLinks);
+    }
+
+
+
+    /**
+     * Validate the existence and permissions of regular file at path.
+     *
+     * If the {@code parentDirPath} is not {@code null}, then setting of missing permissions will
+     * only be done if {@code path} is under {@code parentDirPath}.
+     *
+     * @param context The {@link Context} to get error string.
+     * @param label The optional label for the regular file. This can optionally be {@code null}.
+     * @param filePath The {@code path} for file to validate. Symlinks will not be followed.
+     * @param parentDirPath The optional {@code parent directory path} to restrict operations to.
+     *                      This can optionally be {@code null}. It is not canonicalized and only normalized.
+     * @param permissionsToCheck The 3 character string that contains the "r", "w", "x" or "-" in-order.
+     * @param setPermissions The {@code boolean} that decides if permissions are to be
+     *                              automatically set defined by {@code permissionsToCheck}.
+     * @param setMissingPermissionsOnly The {@code boolean} that decides if only missing permissions
+     *                                  are to be set or if they should be overridden.
+     * @param ignoreErrorsIfPathIsUnderParentDirPath The {@code boolean} that decides if permission
+     *                                               errors are to be ignored if path is under
+     *                                               {@code parentDirPath}.
+     * @return Returns the {@code errmsg} if path is not a regular file, or validating permissions
+     * failed, otherwise {@code null}.
+     */
+    public static String validateRegularFileExistenceAndPermissions(@NonNull final Context context, String label, final String filePath, final String parentDirPath,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 20. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.